app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt

```diff
@@ -151,8 +155,22 @@ fun HikeDetailScreen(
       zoomController.setVisibility(CustomZoomButtonsController.Visibility.NEVER)
       // Enable touch-controls such as pinch to zoom
       setMultiTouchControls(true)
-      // Limit the vertical scrollable area to avoid the user scrolling too far
-      setScrollableAreaLimitLatitude(MapScreen.MAP_MAX_LATITUDE, MapScreen.MAP_MIN_LATITUDE, 0)
+    }
+  }
+
+  // When the map is ready, it will have computed its bounding box
+  mapView.addOnFirstLayoutListener { _, _, _, _, _ ->
+    // Limit the vertical scrollable area to avoid the user scrolling too far from the hike
+    mapView.setScrollableAreaLimitLatitude(
+        min(MapScreen.MAP_MAX_LATITUDE, mapView.boundingBox.latNorth),
+        max(MapScreen.MAP_MIN_LATITUDE, mapView.boundingBox.latSouth),
+        HikeDetailScreen.MAP_BOUNDS_MARGIN)
+    if (route.bounds.maxLon < HikeDetailScreen.MAP_MAX_LONGITUDE ||
+        route.bounds.minLon > HikeDetailScreen.MAP_MIN_LONGITUDE) {
+      mapView.setScrollableAreaLimitLongitude(
+          max(HikeDetailScreen.MAP_MIN_LONGITUDE, mapView.boundingBox.lonWest),
+          min(HikeDetailScreen.MAP_MAX_LONGITUDE, mapView.boundingBox.lonEast),
+          HikeDetailScreen.MAP_BOUNDS_MARGIN)
```

Comment:
Question: How does this manage the case when a hike is on the ±180° line ?

Reply:
It doesn't. Note however that since the map is meant to be centered on a hike, and the `hike.bounds` attribute is a `Bounds` instance of our own making, it is not possible for a hike to cross that line, by design of the `Bounds` data class.

Reply:
Okay thanks!